go/vt/vtgate/planbuilder/testdata/foreignkey_cases.json

```diff
@@ -3463,7 +3463,7 @@
               "Sharded": false
             },
             "FieldQuery": "select col14 from u_tbl1 where 1 != 1",
-            "Query": "select col14 from u_tbl1 where x = 2 and y = 4 lock in share mode",
+            "Query": "select /*+ SET_VAR(foreign_key_checks=OFF) */ col14 from u_tbl1 where x = 2 and y = 4 lock in share mode",
```

Comment:
what does this SET_VAR do here? `foreign_key_checks` isn't even applicable to `select`, right?

Reply:
It is a side effect of having a `LockAndComment` operator. Should not cause any issue with it.